management/server/personal_access_token.go

```diff
@@ -0,0 +1,61 @@
+package server
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"hash/crc32"
+	"math/rand"
+	"time"
+
+	"codeberg.org/ac/base62"
+	"github.com/rs/xid"
+)
+
+// PersonalAccessToken holds all information about a PAT including a hashed version of it for verification
+type PersonalAccessToken struct {
+	ID             string
+	Description    string
+	HashedToken    [32]byte
+	ExpirationDate time.Time
+	// scope could be added in future
+	CreatedBy string
+	CreatedAt time.Time
+	LastUsed  time.Time
```

Comment:
we could display partially the beginning of the token in the UI, so it might make sense to store 3 to 4 chars in plain text. maybe that would demand an increase in the char number

Reply:
We decided to leave this and reconsider at a later point